### DISCUSSION TITLE
Suboptimal picking of a small particle

### DISCUSSION THREAD

**Discussion by LizelleLL** [tbepler/topaz#70] [General]
Hi @tbepler and @alexjnoble 

I have a dataset of ~7000 micrographs of a relatively small particle and am hoping that you can perhaps give me some guidance w.r.t. optimization of the picking parameters. 

The protein is ~140kDa as a monomer and I have observed that a fraction of particles are present as dimers in the same dataset.  Each monomer is composed of 2 homologous globular domains so that it resembles a dumbbell with the side view being two blobs and the top view one blob on the micrographs.  This has obviously made picking very difficult since the particles are barely visible on the raw micrographs. Really excited to see what Topaz can do with this data!

I have trained a Topaz picking model using the 31000 particles from my current ~7 Angstrom resolution map as positive labels.  I tested both denoised and raw micrographs for training and picking. The denoised jobs were for visual inspection of the picks and the raw jobs were used to enable downstream processing in Relion.  

For denoising, I used the unet model and default settings in run_topaz_denoise.py.

These were the parameters I used for the Relion run_topaz_**train**.py script:
--radius 3
--cnn_model resnet8
--numberofparticles 200
--scalefactor 4
--epochs 10
--(the rest were kept at default values)

These were the parameters I changed from the Relion run_topaz_**pick**.py script:
--radius 10 
--model (epoch10.sav from the training job)
--pick_threshold -6
--scalefactor 4
--select_threshold 0
--(the rest were kept at default values)

I have tried to play around with the radius (tried 7, 10, 17 and 20 pixels) but it had almost no effect on the picks which I find kind of strange. I also tried different pick threshold values (tried -6, -2, 0) but again saw almost no change.

It looks like some of the monomer top views (basically just a single small blob of diameter ~60 Angstroms) and potentially some of the monomer side views (2 blobs - diameter ~160 Angstroms) are not picked. Would you agree? 

Below are examples of picking results for 4 micrographs trained & picked either as noisy or denoised.

Do you have any idea how I might optimize the training/picking? I thought that using a small radius such as 7 may help with top views but I am now wondering, could they perhaps be missed due to the 4x downsampling in Topaz?

_Any help at all would be greatly appreciated!_

Regards

Lizelle

**Noisy micrograph 1** (162 particles picked with radius 7 @ 4xbin and pick threshold 0)

![noisymic1](https://user-images.githubusercontent.com/72646966/98124464-e551e800-1ebb-11eb-8408-cef3ddff4e88.JPG)

**Denoised micrograph 1** (126 particles picked with radius 10 @ 4xbin and pick threshold -2)

![denoisedmic1](https://user-images.githubusercontent.com/72646966/98124486-ec78f600-1ebb-11eb-85e4-4fa630ef9bd7.JPG)

**Noisy micrograph 2** (220 particles picked with radius 7 @ 4xbin and pick threshold 0)

![noisymic2](https://user-images.githubusercontent.com/72646966/98124584-0dd9e200-1ebc-11eb-8c6e-01df18f12322.JPG)

**Denoised micrograph 2** (155 particles picked with radius 10 @ 4xbin and pick threshold -2)

![denoisedmic2](https://user-images.githubusercontent.com/72646966/98124641-1f22ee80-1ebc-11eb-8b64-793d842adfbf.JPG)

**Noisy micrograph 3** (235 particles picked with radius 7 @ 4xbin and pick threshold 0)

![noisymic3](https://user-images.githubusercontent.com/72646966/98124830-5d201280-1ebc-11eb-9c1d-e055e1807bf8.JPG)

**Denoised micrograph 3** (129 particles picked with radius 10 @ 4xbin and pick threshold -2)

![denoisedmic3](https://user-images.githubusercontent.com/72646966/98124872-690bd480-1ebc-11eb-9d5a-bb70bd055835.JPG)

**Noisy micrograph 4** (161 particles picked with radius 7 @ 4xbin and pick threshold 0)

![noisymic4](https://user-images.githubusercontent.com/72646966/98124918-73c66980-1ebc-11eb-82e2-6ab0abd6254b.JPG)

**Denoised micrograph 4** (120 particles picked with radius 10 @ 4xbin and pick threshold -2)

![denoisedmic4](https://user-images.githubusercontent.com/72646966/98124947-7f199500-1ebc-11eb-9a2b-733fc2eab730.JPG)

**Comment by alexjnoble** [comment]
Hi Lizelle,

I have several suggestions and observations:

1. Well-curated training particles is more important than many tens of thousands of not-well-curated training particles. I recommend you go through ~100 random picked micrographs and manually curate them. Topaz will learn to pick the way you train it, so you want to only include real particles that are not on junk or aggregated. Pick as many views as you can. Don't worry about picking all particles in the micrographs.
2. Some of your picks are overlapping. You need to increase the extraction radius until you don't get double picks.
3. To get more particles, decrease select_threshold. To get less particles, increase select_threshold.
4. It's not clear to me what percentage of particles in each images are top-views, if any. You can get a quantitative measure of this by doing tomography on a couple holes and looking at the particles in 3D.
5. To get more side views, collect some tilted images like in 10.1038/nmeth.4347

Let me know if these suggestions help!

Best,
-Alex

**Comment by LizelleLL** [comment]
Hi Alex,
Thanks for the helpful reply! I will give your suggestions a try. 

1. I had a quick look at my training particles and do think that they should be improved.  They gave an okayish map before so I just assumed they would be fine for training and never thought of checking them first. 
It's a pity that there isn't a way (yet) to maintain the Relion directory structure of EPU data when running denoising. I will try to copy the curated training coordinates from the denoised micrographs into the correct directories for Relion but in case that fails, would there be any drawback to only using noisy micrographs (in the correct directories) with high defocus values (where I can actually see my particle) for manual selection?

2. I noticed these overlaps and adjusted the radius from 7 to 20 pixels but saw no change in picks. I will try again once I have a new model. In theory, would a radius corresponding to the smallest dimension of an elongated particle allow picking of all views while a radius of the largest dimension may miss the smaller views? 
How should the radius in the training step be optimized? I kept the default of 3pix radius since it is smaller than the particle radius but wasn't sure how to evaluate whether it's optimal.

3. I understand that the select_threshold parameter will select particles based on their scores but don't understand how that differs from the pick_threshold. It may be an obvious thing I am missing but why do you need both? 

4. This sounds like a good idea but my lab work is unfortunately on hold due to covid and I only have this dataset to work on.  I'll keep this in mind for the future.

Thanks again!
Lizelle

**Comment by tbepler** [comment]
I converted this issue to our new discussion section, which is intended to be a better place for questions/discussions like this!

**Comment by alexjnoble** [comment]
Hi Lizelle,

Sorry for the delay.

1. It's anticipated that the upcoming Relion 3.2 will have Topaz built in as its own tab. Hopefully this solves Relion directory structure issues.  Regarding using noisy, high-defocus micrographs: Topaz will learn to pick based on your training particles and training micrographs. If your picks and micrographs are representative, then it shouldn't be a problem. If you are training on high-defocus images and picking on lower defocus images, then it may underperform compared to training on all representative images.
2. You're correct that increasing the extraction radius to the major axis of the particle may miss smaller projections in crowded areas.  Training radius is not a parameter that you can cross-validate. Instead, it may be optimized by training two models with the same parameters except for training radius, then extracting and examining the results.
3. Having two thresholds allows you to filter after picking, which is often useful in my experience.

Best,
-Alex
